cmd/controller-manager/app/controllermanager.go

```diff
@@ -103,9 +103,238 @@ func Run(ctx context.Context, opts *options.Options) error {
 	return nil
 }
 
+// ControllerContext defines the context object for controller
+type ControllerContext struct {
+	Mgr                         controllerruntime.Manager
+	ObjectWatcher               objectwatcher.ObjectWatcher
+	Opts                        *options.Options
+	StopChan                    <-chan struct{}
+	DynamicClientSet            dynamic.Interface
+	OverrideManager             overridemanager.OverrideManager
+	ControlPlaneInformerManager informermanager.SingleClusterInformerManager
+}
+
+// InitFunc is used to launch a particular controller.  It may run additional "should I activate checks".
```

Comment:
> It may run additional "should I activate checks".

What does this mean? (I know it's from Kubernetes, but don't know the meaning.)

Reply:
I make mistakes. I know we just need init controller